src/EFCore.PG/Migrations/NpgsqlMigrationsSqlGenerator.cs

```diff
@@ -880,13 +880,19 @@ protected override void Generate(EnsureSchemaOperation operation, IModel model,
 
             var dbName = DelimitIdentifier(operation.Name);
 
-            builder
-                // TODO: The following revokes connection only for the public role, what about other connecting roles?
-                .AppendLine($"REVOKE CONNECT ON DATABASE {dbName} FROM PUBLIC;")
-                // TODO: For PG <= 9.1, the column name is prodpic, not pid (see http://stackoverflow.com/questions/5408156/how-to-drop-a-postgresql-database-if-there-are-active-connections-to-it)
-                .AppendLine($"SELECT pg_terminate_backend(pg_stat_activity.pid) FROM pg_stat_activity WHERE datname = '{operation.Name}';")
-                .EndCommand(suppressTransaction: true)
-                .AppendLine($"DROP DATABASE {dbName};");
+            if (_postgresVersion.AtLeast(13))
+            {
+                builder.AppendLine($"DROP DATABASE {dbName} WITH (FORCE);");
+            }
+            else
+            {
+                builder
+                    .AppendLine($"REVOKE CONNECT ON DATABASE {dbName} FROM PUBLIC;")
```

Comment:
Should the first comment be preserved?

Reply:
You mean the one about "public role", right? We haven't received a single complaint about this since the project started, and now that PG13 has a better, official way to force-drop a database, I don't think we'll ever actually do anything to improve the older method. Does that sound OK?

Reply:
Yep.